src/runtime/pkg/containerd-shim-v2/create.go

```diff
@@ -355,6 +357,7 @@ func createVmmUser() (string, error) {
 	// Add retries to mitigate temporary errors and race conditions. For example, the user already exists
 	// or another instance of the runtime is also creating a user.
 	maxAttempt := 5
+	rand.Seed(time.Now().UnixNano())
 	for i := 0; i < maxAttempt; i++ {
 		userName = fmt.Sprintf("kata-%v", rand.Intn(100000))
 		_, err = utils.RunCommand([]string{useraddPath, "-M", "-s", nologinPath, userName, "-c", "\"Kata Containers temporary hypervisor user\""})
```

Comment:
`useradd` doesn't really handle races. How is race handled here? It seems possible that a user might still be used for different vmms?

Reply:
I believe `useradd` locks the `/etc/passwd` and `/etc/group` files. Once a user is created and given to a VM, it will not be reused.